### PR TITLE
HPCC-15993 Prevent embedded c++ from reporting other warnings as errors

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review.

File: ecl/hqlcpp/hqlcpp.cpp [hunk 1]
```diff
@@ -1781,7 +1781,7 @@ void HqlCppTranslator::cacheOptions()
         DebugOption(options.optimizeSortAllFieldsStrict,"optimizeSortAllFieldsStrict",false),
         DebugOption(options.alwaysReuseGlobalSpills,"alwaysReuseGlobalSpills",true),
         DebugOption(options.forceAllDatasetsParallel,"forceAllDatasetsParallel",false),  // Purely for regression testing.
-        DebugOption(options.embeddedWarningsAsErrors,"embeddedWarningsAsErrors",true),
+        DebugOption(options.embeddedWarningsAsErrors,"embeddedWarningsAsErrors2",true),
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlcpp.cpp]: EmbeddedWarningsFatal perhaps?

File: ecl/hqlcpp/hqlcpp.cpp [hunk 2]
```diff
@@ -11964,13 +11966,21 @@ void HqlCppTranslator::buildFunctionDefinition(IHqlExpression * funcdef)
...
             {
-                BuildCtx funcctx2(*code, parentHelpersAtom);
-                buildCppFunctionDefinition(funcctx2, bodyCode, proto);
+                if (options.spanMultipleCpp)
+                {
+                    BuildCtx funcctx2(*code, parentHelpersAtom);
+                    buildCppFunctionDefinition(funcctx2, bodyCode, proto);
+                }
+                else
+                    buildCppFunctionDefinition(funcctx, bodyCode, proto);
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlcpp.cpp]: We COULD put inlines at the end still when spanMultipleCpp is off...

**Review by richardkchapman — COMMENTED** [review]
Minor comments

**Comment by ghalliday** [comment]
@richardkchapman changed the error text.  I think I prefer the inline to stay there for all cases.

**Pull request closed without merging**